chromium_src/crypto/hmac.cc

```diff
@@ -0,0 +1,37 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "crypto/hmac.h"
+
+#include <stddef.h>
+
+#include <algorithm>
+#include <string>
+
+#include "base/stl_util.h"
+#include "crypto/openssl_util.h"
+#include "crypto/secure_util.h"
+#include "crypto/symmetric_key.h"
+#include "third_party/boringssl/src/include/openssl/hmac.h"
+
+namespace crypto {
+
+// This function is identical to other Sign() function except it takes a raw
+// buffer and buffer length instead of a StringPiece.
+bool HMAC::Sign(unsigned char* buffer,
```

Comment:
does this method have to be added to the actual hmac class? Can create some kind of utility method instead?

Reply:
or alternatively can we convert the buffer to a stringpiece? I don't believe that anything is copied if you instantiate a new StringPiece from a char*

Reply:
Well we could. It seemed cleaner to extend the class so we don't have to rewrite `Init` from scratch too.

Reply:
`net::x509_util::CryptoBufferAsStringPiece` maybe?